communications/repositories/mongo.py

```diff
@@ -49,6 +66,8 @@ def add_message(self, room_id: str, message: Message):
         message_dict = message.__dict__
```

Comment:
add this code before the "def add_message" method:

        except PyMongoError as e:
            logger.error(f"Failed to retrieve chatroom: {e}", exc_info=True)
            return None

and everything in the "def add_message" method must be wrapped in " try.... except"

for example:
except PyMongoError as e:
            logger.error(f"Failed to add message to MongoDB: {e}", exc_info=True)
        except Exception as e:
            logger.error(f"Unexpected error during message encryption or database update: {e}", exc_info=True)

Reply:
Done